src/Strategy/Python/Poetry/Common.hs

```diff
@@ -154,20 +176,29 @@ toCanonicalName :: Text -> Text
 toCanonicalName t = toLower $ replace "_" "-" (replace "." "-" t)
 
 -- | Maps poetry lock package to map of package name and associated dependency.
-toMap :: [PoetryLockPackage] -> Map.Map PackageName Dependency
-toMap pkgs = Map.fromList $ (\x -> (canonicalPkgName x, toDependency x)) <$> (filter supportedPoetryLockDep pkgs)
+toMap :: [PackageName] -> [PoetryLockPackage] -> Map.Map PackageName Dependency
+toMap prodPkgs pkgs = Map.fromList $ (\x -> (canonicalPkgName x, toDependency x)) <$> (filter supportedPoetryLockDep pkgs)
   where
     canonicalPkgName :: PoetryLockPackage -> PackageName
     canonicalPkgName pkg = PackageName $ toCanonicalName $ unPackageName $ poetryLockPackageName pkg
 
+    canonicalPkgName' :: PackageName -> PackageName
+    canonicalPkgName' = PackageName . toCanonicalName . unPackageName
+
+    canonicalProdPkgNames :: [PackageName]
+    canonicalProdPkgNames = map canonicalPkgName' prodPkgs
+
+    isProductionDirectDep :: PoetryLockPackage -> Bool
+    isProductionDirectDep pkg = canonicalPkgName pkg `elem` canonicalProdPkgNames
```

Comment:
Prefer something like this which is potentially more descriptive than using `'`. Also, Please make `canonicalProdPkgNames` a `Set` since this could potentially be large and expensive to use `elem` with.

```suggestion
     
    canonicalPkgName :: PackageName -> PackageName
    canonicalPkgName = PackageName . toCanonicalName . unPackageName
    
    lockCanonicalPackageName :: PoetryLockPackage -> PackageName
    lockCanonicalPackageName = canonicalPkgName . poetryLockPackageName

    canonicalProdPkgNames :: Set [PackageName]
    canonicalProdPkgNames = Set.fromList . map canonicalPkgName $ prodPkgs

    isProductionDirectDep :: PoetryLockPackage -> Bool
    isProductionDirectDep pkg = canonicalPkgName pkg `Set.member` canonicalProdPkgNames
```